mono/mini/interp/interp.c

```diff
@@ -5437,12 +5445,21 @@ mono_interp_run_filter (StackFrameInfo *frame, MonoException *ex, int clause_ind
 {
 	InterpFrame *iframe = frame->interp_frame;
 	ThreadContext *context = mono_native_tls_get_value (thread_context_id);
+	InterpFrame child_frame;
+	stackval retval;
 
-	gconstpointer ip = iframe->ip;
-	interp_exec_method_full (iframe, context, handler_ip, ex, clause_index);
-	iframe->ip = ip;
-	/* ENDFILTER stores the result into iframe->retval */
-	return iframe->retval->data.i ? TRUE : FALSE;
+	/*
+	 * Have to run the clause in a new frame which is a copy of IFRAME, since
+	 * during debugging, there are two copies of the frame on the stack.
```

Comment:
that's not specific to the debugger, you need to run it in a different frame so it doesn't have any side-effects. that's how it is done in the interpreter exception handling: https://github.com/mono/mono/blob/110888fd661f731814cc7b63be77ebd534e9b759/mono/mini/interp/interp.c#L4992

I think we don't need that extra argument in `interp_exec_method_full`.